pkg/local_object_storage/engine/control.go

```diff
@@ -248,27 +254,31 @@ func (e *StorageEngine) Reload(rcfg ReConfiguration) error {
 		}
 	}
 
-	// mark new shards for addition
+loop:
 	for newID := range rcfg.shards {
-		addShard := true
 		for _, sh := range e.shards {
 			// This calculation should be kept in sync with node
 			// configuration parsing during SIGHUP.
 			if newID == calculateShardID(sh.DumpInfo()) {
-				addShard = false
-				break
+				shardsToReload = append(shardsToReload, reloadInfo{
+					sh:   sh.Shard,
+					opts: rcfg.shards[newID],
+				})
+				continue loop
 			}
 		}
 
-		if addShard {
-			shardsToAdd = append(shardsToAdd, newID)
-		}
+		shardsToAdd = append(shardsToAdd, newID)
 	}
 
 	e.mtx.RUnlock()
 
 	e.removeShards(shardsToRemove...)
 
+	for _, p := range shardsToReload {
```

Comment:
no restrictions on shard's mode? is it safe to reload RW shard?

Reply:
Why not?

Reply:
hm, looked at that one more time. is shard's mutex (implicitly via the `GetMode()`) the only thing that saves us from 1. refilling meta and changing the blobstor 2. closing/opening meta and performing requests to it?

Reply:
There are no problems with _this_ line, `Reload` must be called on each shard which in turn determines whether it needs to perform any changes.

Reply:
The problem already exists in other places, let's fix it separately https://github.com/nspcc-dev/neofs-node/issues/1956